### ISSUE TITLE
Scale out to between 0 and 1 before computing loss function in keras

### ISSUE THREAD

**Issue by anorak94** [keras-team/keras#13973]
Is there a way to scale the output of a CNN before computing the loss function in keras. It would have to work for batches as well. Specifically I am training images and the output is a probability map of each pixels for the entire batch. Then I compute the loss function with the ground truth data. The problem is that the predicted output during training at some intermediate stage lies between 0.02 to 1e-4 and the ground truth pixels are between 0 and 1 so the loss function doesnt penalize pixels which dont look similar to the ground truth. I get this by visualizing the squared error between the output at some intermediate stage and the ground truth data.

![im_test (1)](https://user-images.githubusercontent.com/45297533/79429591-00184a00-7fc8-11ea-8c22-9be63c97b1bc.png)

I would like the squared error between ground truth and prediction to look like this which i can get by scaling the prediction between 0 and 1 and then computing the squared difference of ground truth and prediction

![im_test (2)](https://user-images.githubusercontent.com/45297533/79429617-0c040c00-7fc8-11ea-8f04-8e5a4c136aaf.png)

**Comment by sushreebarsa** [comment]
@anorak94 Moving this issue to closed status as there has been no recent activity, in case you still face the error please create a new issue.Thanks!
